### PR TITLE
Patch sequence to integrate the TF compiler into the CMake build.

### PR DESCRIPTION
**Pull request by stellaraccident**

### CODE REVIEW

File: build_tools/cmake/bazel.bat.in [hunk 1]
```diff
@@ -0,0 +1,3 @@
+@echo off
+cd /d "@_bazel_src_root@"
+@IREE_BAZEL_EXECUTABLE@ @_bazel_startup_options@ %* || exit /b
```
[the remark this answers is not in the document]
Comment (GMNGeoffrey) [reply build_tools/cmake/bazel.bat.in:3]: Yes

File: bindings/python/pyiree/compiler2/tools.py [hunk 2]
```diff
@@ -149,7 +149,9 @@ def invoke_immediate(command_line: List[str],
       run_args["input"] = immediate_input
 
     # Capture output.
-    run_args["capture_output"] = True
+    # Upgrade note: Python >= 3.7 can just use capture_output=True
```
Comment (phoenix-meadowlark) [inline bindings/python/pyiree/compiler2/tools.py:152]: There area few other instances of things like this in the codebase (including another for exactly this issue). I'll throw together a github issue to group them together.

File: bindings/python/README.md [hunk 3]
```diff
@@ -1,19 +1,43 @@
...
+
+## Development mode
+
+For development, just set your `PYTHONPATH` environment variable to the
+`bindings/python` directory in your CMake build dir.
+
+## Run tests
+
+Tests under `bindings/python/tests` can be run directly once installed.
+Additional tests under `integrations/tensorflow/e2e` will be runnable soon.
```
Comment (GMNGeoffrey) [inline bindings/python/README.md:43]: Well I can't comment on the empty file (thanks GitHub), but I believe I mentioned previously that I think it's weird to have two different mechanisms for skipping bazel -> cmake

File: CMakeLists.txt [hunk 4]
```diff
@@ -46,6 +46,7 @@ option(IREE_BUILD_DEBUGGER "Builds the IREE debugger app." OFF)
 option(IREE_BUILD_PYTHON_BINDINGS "Builds the IREE python bindings" OFF)
 option(IREE_BUILD_JAVA_BINDINGS "Builds the IREE java bindings." OFF)
 option(IREE_BUILD_EXPERIMENTAL "Builds experimental projects." OFF)
+option(IREE_BUILD_TENSORFLOW_COMPILER "Builds TensorFlow compiler." OFF)
```
[the remark this answers is not in the document]
Comment (stellaraccident) [reply CMakeLists.txt:49]: Done. That list could use sorting but will do in a followup to not blow up the diff.

File: CMakeLists.txt [hunk 5]
```diff
@@ -371,22 +384,28 @@ if(${IREE_ENABLE_MLIR})
...
 
+#-------------------------------------------------------------------------------
+# Bazel setup (conditional on whether features need it)
+# Depends on python configuration.
+#-------------------------------------------------------------------------------
+
+if(${IREE_BUILD_TENSORFLOW_COMPILER})
+  include(configure_bazel)
+  iree_configure_bazel()
+endif()
```
[the remark this answers is not in the document]
Comment (stellaraccident) [reply CMakeLists.txt:403]: Updated. Can refine more as we go.

File: bindings/python/README.md (hunk 3 above)
Comment (stellaraccident) [reply bindings/python/README.md:43]: Yeah, forgot about that. I can consolidate in a followup. Personally, I find it somewhat... odd... if a CMakeLists.txt has to declare its own intention to not get clobbered: I'd rather have a directory level flag like this. But can do that in another PR.

File: bindings/python/pyiree/compiler2/tools.py (hunk 2 above)
Comment (stellaraccident) [reply bindings/python/pyiree/compiler2/tools.py:152]: +1

File: build_tools/cmake/bazel.bat.in (hunk 1 above)
[the remark this answers is not in the document]
Comment (stellaraccident) [reply build_tools/cmake/bazel.bat.in:3]: Bleh. Updated.

**Review by ScottTodd — APPROVED** [review]

**Pull request closed without merging**